tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -81,23 +81,43 @@ ROOT::Experimental::RNTupleImporter::RLeafArrayTransformation::Transform(const R
 }
 
 ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::RNTupleImporter>>
-ROOT::Experimental::RNTupleImporter::Create(std::string_view sourceFile, std::string_view treeName,
-                                            std::string_view destFile)
+ROOT::Experimental::RNTupleImporter::Create(std::string_view sourceFileName, std::string_view treeName,
+                                            std::string_view destFileName)
 {
    auto importer = std::unique_ptr<RNTupleImporter>(new RNTupleImporter());
    importer->fNTupleName = treeName;
-   importer->fSourceFile = std::unique_ptr<TFile>(TFile::Open(std::string(sourceFile).c_str()));
-   if (!importer->fSourceFile || importer->fSourceFile->IsZombie()) {
-      return R__FAIL("cannot open source file " + std::string(sourceFile));
+   auto sourceFile = TFile::Open(std::string(sourceFileName).c_str());
```

Comment:
I think we need to keep the unique pointer for the TFile source file in the class.  Otherwise we get a resource / memory leak (`TFile::Open` returns a raw pointer to an open file). With the other `Create()` method, the unique pointer would simply stay default-initialized.

Reply:
> The Create factory methods are identical for initializing the destination. They could use a common, private static method to take care of that task.

Since there will already be an importer object, this method does not necessarily have to be static. Is there a benefit to making this method static? It seems to me like that would mostly require passing around more arguments and return values.

Reply:
I think there are two advantages to the factory method:
  - Consistent interface to creating an importer
  - Easier handling of errors when creating the output file. With the factory method, we avoid constructing a "zombie" object or throwing exceptions from the constructor, which can be tricky to do without resource leaks